types/styled-system/index.d.ts

```diff
@@ -36,17 +36,20 @@ export interface styleFn {
     propTypes?: string[];
 }
 
-export interface LowLevelStylefunctionArguments {
-    prop: string;
-    cssProperty?: string;
-    alias?: string;
-    key?: string;
-    transformValue?: (n: string | number, scale: Array<string | number>) => any;
-    scale?: Array<string | number>;
+export type Scale = object | Array<string | number>;
+
+export interface LowLevelStylefunctionArguments<N> {
+  prop: string;
+  cssProperty?: string;
+  alias?: string;
+  key?: string;
+  transformValue?: (n: N, scale?: Scale) => any;
```

Comment:
Can you make this generic on `LowLevelStylefunctionArguments<N, S = Scale>`?

Reply:
Test could be my previous example:

```ts
const buttonSizes = {
  sm: '8px',
  md: '12px',
  lg: '24px',
};

const buttonSizeWithGeneric = style<keyof buttonSizes, typeof buttonSizes>({
  prop: 'size',
  cssProperty: 'font-size',
  transformValue: (size, sizes) => sizes[size]
});
```

Reply:
Will do!

Reply:
@StevenLangbroek I updated this PR to make scale a generic. I had to do it on `style`, the same way I did it for N, but it should work. Let me know if you're good with this change!